diff --git a/lib/uninorm/composition.c b/lib/uninorm/composition.c
index 645099e87a..c97866a634 100644
--- a/lib/uninorm/composition.c
+++ b/lib/uninorm/composition.c
@@ -70,12 +70,12 @@ uc_composition (ucs4_t uc1, ucs4_t uc2)
           char codes[6];
           const struct composition_rule *rule;
 
-          codes[0] = (uc1 >> 16) & 0xff;
-          codes[1] = (uc1 >> 8) & 0xff;
-          codes[2] = uc1 & 0xff;
-          codes[3] = (uc2 >> 16) & 0xff;
-          codes[4] = (uc2 >> 8) & 0xff;
-          codes[5] = uc2 & 0xff;
+          codes[0] = (char) ((uc1 >> 16) & 0xff);
+          codes[1] = (char) ((uc1 >> 8) & 0xff);
+          codes[2] = (char) (uc1 & 0xff);
+          codes[3] = (char) ((uc2 >> 16) & 0xff);
+          codes[4] = (char) ((uc2 >> 8) & 0xff);
+          codes[5] = (char) (uc2 & 0xff);
 
           rule = gl_uninorm_compose_lookup (codes, 6);
