diff --git a/build-aux/gnu-web-doc-update b/build-aux/gnu-web-doc-update
index a804031a43..8c901575d8 100755
--- a/build-aux/gnu-web-doc-update
+++ b/build-aux/gnu-web-doc-update
@@ -1,6 +1,7 @@
 #!/bin/sh
 # Run this after each non-alpha release, to update the web documentation at
-# https://www.gnu.org/software/$pkg/manual/
+# https://www.gnu.org/software/libidn/libidn2/manual/
+pkg="libidn/libidn2"
 
 VERSION=2023-03-23.02; # UTC
 
